nilearn/image/image.py

```diff
@@ -1279,6 +1285,36 @@ def math_img(formula, copy_header_from=None, **imgs):
     in FSL.
 
     """
+    is_surface = all(isinstance(x, SurfaceImage) for x in imgs.values())
+
+    if is_surface:
+        first_img = next(iter(imgs.values()))
+        for image in imgs.values():
+            check_polymesh_equal(first_img.mesh, image.mesh)
```

Comment:
I think this might throw exception in case the meshes are not equal. It might be good to catch the exception and display a corresponding message to user.

Reply:
> isn't that what the function already does:

Yes, it does :) I had thought some message more specific to this function. But not really necessary may be.

Reply:
> here I have assumed that it is enough for the meshes to be compatible (same number of vertices), even if they do not have exactly the same geometry

I cannot comment on if we should be more strict or not, however in docstring it might be better to clearly indicate what is expected.

In that case, the exception message to user can be made more clear with the expected equality of meshes.

Reply:
actually I will make it more strict: if users complain we can revisit it later